pennylane/devices/null_qubit.py

```diff
@@ -221,15 +230,38 @@ def name(self):
         """The name of the device."""
         return "null.qubit"
 
-    def __init__(self, wires=None, shots=None) -> None:
+    def __init__(
+        self,
+        wires=None,
+        shots=None,
+        target_device=None,
+        operations=None,
+        assume_no_broadcasting=False,
+    ) -> None:
         super().__init__(wires=wires, shots=shots)
         self._debugger = None
+        self._assume_no_broadcasting = assume_no_broadcasting
+        if target_device and not isinstance(target_device, Device):
+            raise NotImplementedError(
+                "null.qubit only supports target devices that obey the new device interface"
+            )
+        if target_device and operations:
+            raise NotImplementedError(
+                "null.qubit accepts either target_device or operations, but not both."
+            )
+
+        self._target_device = target_device
+        self._operations = operations
+
+    def _get_batch_size(self, circuit):
+        return None if self._assume_no_broadcasting else circuit.batch_size
 
     def _simulate(self, circuit, interface):
         shots = circuit.shots
         obj_with_wires = self if self.wires else circuit
+
         results = tuple(
-            zero_measurement(mp, obj_with_wires, shots, circuit.batch_size, interface)
+            zero_measurement(mp, obj_with_wires, shots, self._get_batch_size(circuit), interface)
```

Comment:
I remember you mentioning computing `batch_size` taking time? Should we then compute it just once and use it here for `results` and below for `derivatives`?